### ISSUE TITLE
empty test coverage

### ISSUE THREAD

**Issue by jamestalmage** [bcoe/nyc#52]
see https://github.com/jamestalmage/promisify-firebase

changing nothing but `index.js` and `test.js` to a minimal test case (i.e., not changing any of the build config / package.json / etc), I can get working coverage. This means that the code of my module itself is somehow disabling the coverage.

I've poked around trying to figure out exactly what line is causing it, so far no luck.

any help would be appreciated.

**Comment by jamestalmage** [comment]
So I have narrowed things down quite a bit. See [here](https://github.com/jamestalmage/promisify-firebase/blob/minimal-nyc-working/test.js#L19-L21). 

Simply instantiating a new `Firebase` disables `nyc`:

```js
// uncomment the following and nyc is broken:
// new Firebase('https://test.firebaseio.test');
```

The debug source for Firebase is [here](https://github.com/firebase/firebase-bower/blob/master/firebase-debug.js), unfortunately that is just the web build, which is slightly different than the node build. I do not know if a debug build for node is available.

My only guess is that Firebase is somehow interfering with process forking and/or exiting (code that likely is not in the web build linked above). Hopefully one of the firebase folks can lend some insight  (cc: @jwngr, @katowulf)

**Comment by bcoe** [comment]
@jamestalmage thanks for digging into this! I'm on my honeymoon right now so my turn around on open-source is a bit slow, but I'm hoping to dig into some of the open issues on nyc ASAP.

**Comment by jamestalmage** [comment]
Congratulations!

Don't give it another thought until you are back. I'll live.

**Comment by jwngr** [comment]
Firebase has a known issue where the process doesn't die until you explicitly call `process.exit()`.  But I don't understand this issue well enough / have the right context to know if that's related to this issue.

**Comment by jamestalmage** [comment]
Even if you call `Firebase.goOffline()`?

**Comment by jwngr** [comment]
Yeah, this is a recent issue. The Google Groups thread is from a long while ago. @mikelehen tells me this is still an ongoing issue, I believe even if you call `Firebase.goOffline()`.

**Comment by jamestalmage** [comment]
Turns out to be a combination of the Firebase long running process, and an issue with the way `ava` closes forked processes.

It was killing the parent process, which means the `onExit` listeners for each forked process were never triggered. Consequently the coverage data was never written to disk.

I have filed a PR that will fix this in `ava`: https://github.com/sindresorhus/ava/issues/120

Perhaps there is a way for NYC to track forked processes and automatically round up existing data when the parent process is killed.
